dasharo-compatibility/esp-scanning.robot

```diff
@@ -0,0 +1,98 @@
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+ESP001.001 ESP Scan with OS-specific .efi files added
+    [Documentation]    This test aims to verify that any properly added .efi
+    ...    files will have boot menu entries created for them.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP001.001 not supported
+
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+    Login To Linux
+    Switch To Root User
+    Remove All Supported Systems From Efi
+    Populate /EFI With System Folders
+    Power On
+    Enter Boot Menu Tianocore
+    Check Boot Menu For All Supported Systems    normal
+
+ESP002.001 ESP Scan after deleting additional .efi files
+    [Documentation]    This test aims to verify that none of the systems linger
+    ...    on in the boot menu after we've deleted their files from /EFI/.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP002.001 not supported
+
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+    Login To Linux
+    Switch To Root User
+    Remove All Supported Systems From Efi
+    Power On
+    Enter Boot Menu Tianocore
+    Check Boot Menu For All Supported Systems    empty
+
+ESP003.001 ESP Scan ignores OSes on removable media
+    [Documentation]    This test aims to verify that the bootable /EFI
+    ...    partitions of removable media are ignored by the scan and aren't
+    ...    listed in boot menu, except for DTS.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP003.001 not supported
+
+    Upload And Mount Tinycore
+    Power On
+    Enter Boot Menu Tianocore
+    ${result}=    Run Keyword And Return Status
+    ...    Check That USB Devices Are Detected    ${TRUE}
+    Should Not Be True    ${result}
+
+ESP004.001 ESP Scan does not create duplicate entries
+    [Documentation]    This test aims to verify that the firmware will not
+    ...    create duplicate entries, for example, if both shimx64 and grubx64
+    ...    are present for a single OS.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP004.001 not supported
+
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+    Login To Linux
+    Switch To Root User
+    Populate /EFI With System Folders
+    Power On
+    Enter Boot Menu Tianocore
+    Check Boot Menu For All Supported Systems    double_entry_check
+
+ESP005.001 ESP Scan detects Dasharo Tools Suite
+    [Documentation]    This test aims to verify that the firmware detects
+    ...    Dasharo Tools Suite boot media and creates a corresponding boot
+    ...    menu entry.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ESP005.001 not supported
+
+    Upload And Mount DTS Flash ISO
+    Power On
+    Enter Boot Menu Tianocore
+    Check That USB Devices Are Detected    ${TRUE}
```

Comment:
Where is the check for `Dasharo Tools Suite (on <disk_name>)`?

Reply:
@miczyg1 Right - IIRC on removable media DTS doesn't get listed as `Dasharo Tools Suite (on <disk_name>)` like it does on the SSD, but gets listed as the flash drive brand name or `PiKVM Composite Device`. I should rather report this as a bug than make the test go with it.

Reply:
@miczyg1 I rewrote the check for DTS [here](https://github.com/Dasharo/open-source-firmware-validation/pull/31/files#diff-c8f7a5cdc7c64ba736049ef48cf813ae918ab342659418e21d720981c53f903aR99). It always fails due to what I mentioned above. AFAIK the esp-scan was a feature merged to our edk2 fork - I wanted to report this issue, but there is no issues tab under edk2. Where should I report the issue then?

Reply:
I think @TomaszAIR didn't fix it yet. The DTS boot partition is standard FAT32, not EFI System Partition type. That is why it cannot be found.